pkg/events/eifactory/factory.go

```diff
@@ -18,40 +18,49 @@ package eifactory
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/hyperledger/firefly-common/pkg/config"
 	"github.com/hyperledger/firefly-common/pkg/i18n"
+	"github.com/hyperledger/firefly-common/pkg/log"
 	"github.com/hyperledger/firefly/internal/coremsgs"
 	"github.com/hyperledger/firefly/internal/events/system"
 	"github.com/hyperledger/firefly/internal/events/webhooks"
 	"github.com/hyperledger/firefly/internal/events/websockets"
 	"github.com/hyperledger/firefly/pkg/events"
 )
 
-var plugins = []events.Plugin{
-	&websockets.WebSockets{},
-	&webhooks.WebHooks{},
-	&system.Events{},
+var websocketsPlugin = websockets.WebSockets{}
+var webhooksPlugin = webhooks.WebHooks{}
+var systemEventsPlugin = system.Events{}
+var pluginsByName = map[string]func() events.Plugin{
+	websockets.Name(): func() events.Plugin { return &websocketsPlugin },
+	webhooks.Name():   func() events.Plugin { return &webhooksPlugin },
+	system.Name():     func() events.Plugin { return &systemEventsPlugin },
 }
 
-var pluginsByName = make(map[string]events.Plugin)
+// Allows other code to register additional implementations of Event plugins
+// that can also be initialized with this factory
+func RegisterPlugins(plugins map[string]func() events.Plugin) {
```

Comment:
I think this should be a singleton, where the `Name()` function of the plugin is used to determine the name

Reply:
Ah, ok - so the interface here is to register a new factory for the plugin.
Because we need to be able to generate lots of instances of it.

In which case, I think we should define an `EventPluginFactory` interface, and have `Name()` as a function on that interface, as well as `NewInstance()` or similar.